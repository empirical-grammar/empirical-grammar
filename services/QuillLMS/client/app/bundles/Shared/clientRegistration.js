import 'lazysizes';
import 'lazysizes/plugins/parent-fit/ls.parent-fit';
import ReactOnRails from 'react-on-rails';
import './styles/styles.scss'

import {
  QuestionList,
  LinkListItem,
  QuestionListByConcept,
  Instructions,
  WarningDialogue,
  Prompt,
  ConceptExplanation,
  ArchivedButton,
  Card,
  CarouselAnimation,
  DataTable,
  ProgressBar,
  DropdownInput,
  Error,
  FlagDropdown,
  Input,
  Passthrough,
  ResumeOrBeginButton,
  SmartSpinner,
  Snackbar,
  defaultSnackbarTimeout,
  Spinner,
  TextEditor,
  TextArea,
  Tooltip,
  UploadOptimalResponses,
  Modal,
  ListItem,
  SortableList,
  AffectedResponse,
  PieChart,
  QuestionBar,
  ResponseSortFields,
  ResponseToggleFields,
  Feedback,
  Cue,
  CueExplanation,
  ThankYou,
  SentenceFragments,
  QuestionRow,
  MultipleChoice,
  Register,
  PlayTitleCard,
  TitleCard,
  hashToCollection,
  isValidRegex
} from './index'

ReactOnRails.register({
  QuestionList,
  LinkListItem,
  QuestionListByConcept,
  Instructions,
  WarningDialogue,
  Prompt,
  ConceptExplanation,
  ArchivedButton,
  Card,
  CarouselAnimation,
  DataTable,
  ProgressBar,
  DropdownInput,
  Error,
  FlagDropdown,
  Input,
  Passthrough,
  ResumeOrBeginButton,
  SmartSpinner,
  Snackbar,
  defaultSnackbarTimeout,
  Spinner,
  TextEditor,
  TextArea,
  Tooltip,
  UploadOptimalResponses,
  Modal,
  ListItem,
  SortableList,
  AffectedResponse,
  PieChart,
  QuestionBar,
  ResponseSortFields,
  ResponseToggleFields,
  Feedback,
  Cue,
  CueExplanation,
  ThankYou,
  SentenceFragments,
  QuestionRow,
  MultipleChoice,
  Register,
  PlayTitleCard,
  TitleCard,
  hashToCollection,
  isValidRegex
});
