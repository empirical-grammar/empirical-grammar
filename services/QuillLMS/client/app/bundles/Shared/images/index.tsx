export const bigCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-check-big.svg`,
  alt: "White checkmark in green circle"
}

export const searchMapIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-map-search.svg`,
  alt: "Map with a magnifying glass icon"
}

export const clipboardCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-clipboard-check.svg`,
  alt: "Clipboard with a checkmark icon"
}

export const tableCheckIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-table-check.svg`,
  alt: "Table with a checkmark icon"
}

export const accountViewIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-account-view.svg`,
  alt: "Avatar with an eye next to it icon"
}

export const giftIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-gift.svg`,
  alt: "Gift icon"
}

export const groupAccountIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-account-group.svg`,
  alt: "Three avatars icon"
}

export const googleClassroomIcon = {
  src: `${process.env.CDN_URL}/images/icons/icons-google-classroom.svg`,
  alt: "Google Classroom icon"
}
