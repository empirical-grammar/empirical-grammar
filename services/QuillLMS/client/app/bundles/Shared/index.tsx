export {
  QuestionList,
  LinkListItem,
  QuestionListByConcept
} from './components/questionList/index'

export {
  Instructions,
  Prompt
} from './components/fillInBlank/index'

export {
  ConceptExplanation,
} from './components/feedback/conceptExplanation'

export {
  ArchivedButton,
  Card,
  CarouselAnimation,
  DataTable,
  ProgressBar,
  DropdownInput,
  Error,
  FlagDropdown,
  Input,
  Passthrough,
  ResumeOrBeginButton,
  SmartSpinner,
  Snackbar,
  defaultSnackbarTimeout,
  Spinner,
  SortableList,
  TeacherPreviewMenu,
  TextEditor,
  TextArea,
  Tooltip,
  TwoThumbSlider,
  UploadOptimalResponses
} from './components/shared/index'

export {
  Modal
} from './components/modal/modal'

export {
  AffectedResponse,
  PieChart,
  QuestionBar,
  ResponseSortFields,
  ResponseToggleFields
} from './components/questions/index'

export {
  Feedback,
  Cue,
  CueExplanation,
  ThankYou,
  SentenceFragments
} from './components/renderForQuestions/index'

export {
  QuestionRow
} from './components/scoreAnalysis/index'

export {
  MultipleChoice,
  Register,
  PlayTitleCard
} from './components/studentLessons/index'

export {
  TitleCard
} from './components/titleCards/index'

export {
  hashToCollection,
  isValidRegex,
  momentFormatConstants,
  copyToClipboard
} from './libs/index'
