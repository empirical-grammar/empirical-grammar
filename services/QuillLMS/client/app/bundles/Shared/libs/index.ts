export { hashToCollection } from './hashToCollection'
export { isValidRegex } from './isValidRegex'
export { momentFormatConstants } from './momentFormatConstants'
export { copyToClipboard } from './copyToClipboard'
