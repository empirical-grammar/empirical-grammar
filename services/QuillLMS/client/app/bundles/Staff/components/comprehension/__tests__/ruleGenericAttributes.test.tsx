import * as React from 'react';
import { shallow } from 'enzyme';

import RuleGenericAttributes from '../configureRules/ruleGenericAttributes';
import { DropdownInput, Input, TextEditor } from '../../../../Shared/index'

const mockProps = {
  isUniversal: false,
  errors: {},
  ruleConceptUID: 'test-id',
  ruleDescription: 'test description',
  ruleID: 17,
  ruleName: 'generic rule 17',
  ruleOptimal: false,
  ruleType: { value: 'plagiarism', label: 'Plagiarism'},
  setRuleConceptUID: jest.fn(),
  setRuleDescription: jest.fn(),
  setRuleName: jest.fn(),
  setRuleOptimal: jest.fn(),
  setRuleType: jest.fn()
};

describe('RuleGenericAttributes component', () => {
  let container = shallow(<RuleGenericAttributes {...mockProps} />);

  it('should render RuleGenericAttributes', () => {
    expect(container).toMatchSnapshot();
  });

  it('should render the appropriate form components', () => {
    // TextEditor: Rule Description (1)
    // Input: Name, Concept UID (2)
    // DropdownInput: Type, Description (2)
    expect(container.find(TextEditor).length).toEqual(1);
    expect(container.find(Input).length).toEqual(2);
    expect(container.find(DropdownInput).length).toEqual(2);
  });
});
