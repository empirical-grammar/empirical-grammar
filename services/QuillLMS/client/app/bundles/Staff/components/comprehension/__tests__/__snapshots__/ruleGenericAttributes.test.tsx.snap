// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleGenericAttributes component should render RuleGenericAttributes 1`] = `
<Fragment>
  <DropdownInput
    className="rule-type-input disabled"
    disabled={true}
    handleChange={[Function]}
    isSearchable={true}
    label="Rule Type"
    options={
      Array [
        Object {
          "label": "Regex",
          "value": "rules-based",
        },
        Object {
          "label": "Plagiarism",
          "value": "plagiarism",
        },
      ]
    }
    value={
      Object {
        "label": "Plagiarism",
        "value": "plagiarism",
      }
    }
  />
  <Input
    autoComplete="on"
    className="name-input"
    handleChange={[Function]}
    label="Name"
    value="generic rule 17"
  />
  <Input
    autoComplete="on"
    className="concept-uid-input"
    handleChange={[Function]}
    label="Concept UID"
    value="test-id"
  />
  <DropdownInput
    className="rule-type-input"
    handleChange={[Function]}
    isSearchable={true}
    label="Optimal?"
    options={
      Array [
        Object {
          "label": "Optimal",
          "value": "optimal",
        },
        Object {
          "label": "Sub-Optimal",
          "value": "",
        },
      ]
    }
    value={false}
  />
  <p
    className="form-subsection-label"
  >
    Rule Description
  </p>
  <TextEditor
    ContentState={[Function]}
    EditorState={[Function]}
    handleTextChange={[Function]}
    key="rule-description"
    text="test description"
  />
</Fragment>
`;
